### ISSUE TITLE
xywh, Incorrect coordinates

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#2293]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

results[0].boxes

```
boxes: tensor([[630.0665, 305.6620, 721.4636, 379.8742,   0.8820,   0.0000]], device='cuda:0')
cls: tensor([0.], device='cuda:0')
conf: tensor([0.8820], device='cuda:0')
data: tensor([[630.0665, 305.6620, 721.4636, 379.8742,   0.8820,   0.0000]], device='cuda:0')
id: None
is_track: False
orig_shape: tensor([1024, 1280], device='cuda:0')
shape: torch.Size([1, 6])
xywh: tensor([[675.7650, 342.7681,  91.3971,  74.2122]], device='cuda:0')
xywhn: tensor([[0.5279, 0.3347, 0.0714, 0.0725]], device='cuda:0')
xyxy: tensor([[630.0665, 305.6620, 721.4636, 379.8742]], device='cuda:0')
xyxyn: tensor([[0.4922, 0.2985, 0.5636, 0.3710]], device='cuda:0')
```

X = 630.0665
Y = 305.6620

but ...

xywh: tensor([[675.7650, 342.7681,  91.3971,  74.2122]], device='cuda:0')
XY
xywh   Incorrect coordinates

**Comment by glenn-jocher** [comment]
@monkeycc hi there, 

Based on the code snippet you provided, it seems that you are querying the coordinates of a bounding box object detected by YOLOv8. The coordinate values that you are receiving are in the format of 'x1, y1, x2, y2' which corresponds to 'xmin, ymin, xmax, ymax' respectively. 

However, you seem to have a concern with the xywh (center-x, center-y, width, height) output of the object. Can you please provide more information about your concern? Please note that the center-x and center-y are calculated based on the 'x1, y1, x2, y2' coordinates. 

Feel free to provide more information or ask additional questions if needed. 

Best regards.

**Comment by monkeycc** [comment]
```
boxes: tensor([[630.0665, 305.6620, 721.4636, 379.8742,   0.8820,   0.0000]], device='cuda:0')
cls: tensor([0.], device='cuda:0')
conf: tensor([0.8820], device='cuda:0')
data: tensor([[630.0665, 305.6620, 721.4636, 379.8742,   0.8820,   0.0000]], device='cuda:0')
id: None
is_track: False
orig_shape: tensor([1024, 1280], device='cuda:0')
shape: torch.Size([1, 6])
xywh: tensor([[675.7650, 342.7681,  91.3971,  74.2122]], device='cuda:0')
xywhn: tensor([[0.5279, 0.3347, 0.0714, 0.0725]], device='cuda:0')
xyxy: tensor([[630.0665, 305.6620, 721.4636, 379.8742]], device='cuda:0')
xyxyn: tensor([[0.4922, 0.2985, 0.5636, 0.3710]], device='cuda:0')
```


```python
def xyxy_to_xywh(xyxy_coords):
    xmin, ymin, xmax, ymax = xyxy_coords
    x = xmin
    y = ymin
    w = xmax - xmin
    h = ymax - ymin
    return x, y, w, h




boxes = results[0].boxes

boxes_xywh = boxes.xyxy.cpu().numpy()

for box in boxes_xywh:
    x, y, w, h = xyxy_to_xywh(box[:4]) 
    #  630.0665, 305.6620, 91.3971,  74.2122     This is the correct data

    # but    xywh: tensor([[675.7650, 342.7681,  91.3971,  74.2122]], device='cuda:0')
    # The coordinates of x and y are incorrect

    x, y, w, h = int(x), int(y), int(w), int(h)
    cropped_image = image[y:y + h, x:x + w]


```

`xywh (center-x, center-y, width, height)
`
There is a problem with cropping the image
The coordinates of x and y are incorrect
@glenn-jocher

**Comment by Laughing-q** [comment]
@monkeycc hi, actually that's how we designed, the `xy` in `xywh` is center x and center y, not the left-top coordinates.
refer to:
[https://github.com/ultralytics/ultralytics/blob/243fc4b1fe214ff6c27759dad51c37809db8f7f8/ultralytics/yolo/utils/ops.py#L352-L355](https://github.com/ultralytics/ultralytics/blob/243fc4b1fe214ff6c27759dad51c37809db8f7f8/ultralytics/yolo/utils/ops.py#L352-L355)
But yes the name of it might confuse you, probably we should consider to rename it to something like `cxcywh`.

**Comment by monkeycc** [comment]
Oh, I see
Thank you for your reply


It is indeed recommended to modify the name
`cxcywh`

Also, it is recommended to keep xywh
Modify Results  left-top coordinates.
Because cutting images requires xywh
Of course, this is just a suggestion

**Comment by glenn-jocher** [comment]
@monkeycc thank you for your observation and suggestion regarding the naming convention of the `xywh` format. We agree that the current name might be confusing for some users, and we will consider renaming it to something more descriptive such as `cxcywh` to better reflect the meaning of each value in the format.

Regarding your suggestion to modify the Results to return left-top coordinates instead of `xywh`, we appreciate the input, and we understand that this might be helpful in certain cases. However, we decided to keep `xywh` as it provides a more standardized and concise format for object detection tasks. In addition, converting `xyxy` to `xywh` is a simple and common operation that can be easily implemented using the `xyxy_to_xywh()` function that we provide in our codebase.

**Comment by rathaROG** [comment]
> Oh, I see Thank you for your reply
> 
> It is indeed recommended to modify the name `cxcywh`
> 
> Also, it is recommended to keep xywh Modify Results left-top coordinates. Because cutting images requires xywh Of course, this is just a suggestion

+1

**Comment by glenn-jocher** [comment]
@rathaROG thank you for your feedback and input regarding the naming convention of the `xywh` format. We  acknowledge that the current name might be confusing to some users, and we will consider renaming it to something more descriptive such as `cxcywh` to provide a clearer representation of the meaning of each value in the format.

Regarding your suggestion to modify the Results left-top coordinates, thank you for the suggestion. However, we have decided to keep the `xywh` format as it provides a standardized and concise format for object detection tasks. Nonetheless, we have included helper functions such as `xyxy_to_xywh()` in our codebase to enable simple conversion from `xyxy` to `xywh` format, which is useful for image cropping.
